include/ginkgo/extensions/kokkos/spaces.hpp

```diff
@@ -0,0 +1,280 @@
+                                   MemorySpace>::accessible;
+};
+#endif
+
+
+/**
+ * Checks if the memory space is accessible by the executor
+ *
+ * @tparam MemorySpace  A Kokkos memory space type
+ * @tparam ExecType  A Ginkgo executor type
+ *
+ * @return  true if the memory space is accessible by the executor
+ */
+template <typename MemorySpace, typename ExecType>
+inline bool check_compatibility(std::shared_ptr<const ExecType>)
+{
+    return detail::compatible_space<MemorySpace, ExecType>::value;
+}
+
+
+/**
+ * Checks if the memory space is accessible by the executor
+ *
+ * @tparam MemorySpace  A Kokkos memory space
+ * @param exec  The executor which should access the memory space
+ *
+ * @return  true if the memory space is accessible by the executor
+ */
+
+template <typename MemorySpace>
+inline bool check_compatibility(std::shared_ptr<const Executor> exec)
+{
+    if (auto p = std::dynamic_pointer_cast<const ReferenceExecutor>(exec)) {
+        return check_compatibility<MemorySpace>(p);
+    }
+    if (auto p = std::dynamic_pointer_cast<const OmpExecutor>(exec)) {
+        return check_compatibility<MemorySpace>(p);
+    }
+    if (auto p = std::dynamic_pointer_cast<const CudaExecutor>(exec)) {
+        return check_compatibility<MemorySpace>(p);
+    }
+    if (auto p = std::dynamic_pointer_cast<const HipExecutor>(exec)) {
+        return check_compatibility<MemorySpace>(p);
+    }
+    if (auto p = std::dynamic_pointer_cast<const DpcppExecutor>(exec)) {
+        return check_compatibility<MemorySpace>(p);
+    }
+    GKO_NOT_IMPLEMENTED;
+}
+
+
+/**
+ * Throws if the memory space is ~not~ accessible by the executor associated
+ * with the passed in Ginkgo object.
+ *
+ * @tparam MemorySpace  A Kokkos memory space type..
+ * @tparam T  A Ginkgo type that has the member function `get_executor`.
+ *
+ * @param obj  Object which executor is used to check the access to the memory
+ * space.
+ */
+template <typename MemorySpace, typename T>
+inline void assert_compatibility(T&& obj)
+{
+    GKO_THROW_IF_INVALID(check_compatibility<MemorySpace>(obj.get_executor()),
+                         "Executor type and memory space are incompatible");
+}
+
+
+}  // namespace detail
+
+
+/**
+ * Creates an Executor matching the Kokkos::DefaultHostExecutionSpace.
+ *
+ * If no kokkos host execution space is enabled, this throws an exception.
+ *
+ * @return  An executor of type either ReferenceExecutor or OmpExecutor.
+ */
+inline std::shared_ptr<Executor> create_default_host_executor()
+{
+#ifdef KOKKOS_ENABLE_SERIAL
+    if constexpr (std::is_same_v<Kokkos::DefaultHostExecutionSpace,
+                                 Kokkos::Serial>) {
+        return ReferenceExecutor::create();
+    }
+#endif
+#ifdef KOKKOS_ENABLE_OPENMP
+    if constexpr (std::is_same_v<Kokkos::DefaultHostExecutionSpace,
+                                 Kokkos::OpenMP>) {
+        return OmpExecutor::create();
+    }
+#endif
```

Comment:
If both are available, only one of them will be the `DefaultHostExecutionSpace`. Kokkos will always use `OpenMP` before `Serial`, so this will pick the `OmpExecutor`.